### ISSUE TITLE
Cylc Install, nested source directory

### ISSUE THREAD

**Issue by datamel** [cylc/cylc-flow#4453]
Currently, you can perform cylc install workflow1 on a source directory with the following structure...
```
   $ tree ~/cylc-src
   └── workflow1
       ├── workflow2
       │   └── flow.cylc 
       └── flow.cylc
```

Cylc install does check for nested _run_ directories (by looking for `.service` directory) but this current scenario will install without error. 

Cylc scan, will not currently pick up `workflow2` (`workflow1` does appear on scan).

Some thought needed into whether we wish to check and forbid a scenario like this (although may be undesirable for performance to nested scan), or perhaps clear documentation on best practice for sub-workflows and/or a tutorial would be an acceptable solution.

**Comment by hjoliver** [comment]
(My comments above might be slightly off-piste, if you're only worried about `cylc scan` seeing these source dirs, or having `cylc install` balk at them?)

**Comment by hjoliver** [comment]
Closed by #4861 ... 

...  The nested source structure in the description above is the natural way to store sub-workflow source directories, and #4861 makes this work in Cylc 8 by allowing source directories under `cylc-run` (for on-the-fly installation of sub-workflow instances).
